processor.go

```diff
@@ -523,10 +523,27 @@ func (g *Processor) newStorage(topic string, id int32, update UpdateCallback) (*
 		}, nil
 	}
 
-	st, err := g.opts.builders.storage(topic, id)
+	wg := new(sync.WaitGroup)
```

Comment:
Don't know why, but usually I'd create a WaitGroup not as a pointer, so doing `var wg sync.WaitGroup` instead.
But that's just personal flavor. Either way you could then use the multiline-var with `var ( ... ) ` for the other variables below too.

Reply:
I agree, don't allocate the wait group if its lifetime is only within this function scope. That would only create confusion for those reading the code.